src/report/sagas.js

```diff
@@ -0,0 +1,36 @@
+
+import { all, call, delay, put, takeEvery } from 'redux-saga/effects';
+import LoggingService from '@edx/frontend-logging';
+
+
+// Actions
+import {
+  FETCH_REPORTS,
+  fetchReportsBegin,
+  fetchReportsFailure,
+  fetchReportsSuccess,
+} from './actions';
+
+// Services
+import * as ApiService from './service';
+
+
+export function* handleFetchReports({ payload: { programKey }}) {
+  try {
+    yield put(fetchReportsBegin());
+
+    const data = yield call(ApiService.getReportsByProgram, programKey);
+    if (data.length > 0) {
+      yield put(fetchReportsSuccess(programKey, data));
+    } else {
+      yield put(notAuthenticated());
```

Comment:
I don't think not having reports should be considered 'not authenticated' since the existence of files isn't determining access like the /programs call on the upload page.  If anything it would be an error or just considered empty.